core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -44,7 +44,7 @@ protected DataTableScan(
     this(table, schema, context);
   }
 
-  protected DataTableScan(Table table, Schema schema, TableScanContext context) {
+  public DataTableScan(Table table, Schema schema, TableScanContext context) {
```

Comment:
I don't think that this should be public. If you need to have stats returned, then that option needs to be exposed through a public interface, not on the implementation class.

Reply:
Reverted